test/functional/test_runner.py

```diff
@@ -559,6 +562,10 @@ def run_tests(*, test_list, src_dir, build_dir, tmpdir, jobs=1, enable_coverage=
     if os.path.isdir(cache_dir):
         print("%sWARNING!%s There is a cache directory here: %s. If tests fail unexpectedly, try deleting the cache directory." % (BOLD[1], BOLD[0], cache_dir))
 
+    # Exit the test if there is not enough space on the testing dir
+    if shutil.disk_usage(tmpdir).free < MIN_FREE_SPACE:
+        sys.exit(f"Tests aborted: Insufficient space available in directory: {tmpdir}")
```

Comment:
Since the `MIN_FREE_SPACE` value is a bit of a guess, it might make sense to either make this into a VERY LOUD non-fatal warning, or add an arg like `--non-fatal-disk-space-check` to make it into a warning. That way people running in a bit resource-constrained environments and/or only interested in a specific low-disk-usage test don't need to edit the `MIN_FREE_SPACE` to be able to run their tests.

Reply:
@maflcko I have converted this to a warning as suggested